File: src/contests/cg/puzzle/java/the-river-i-.java
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

import java.io.*;

class Solution {
    static long digitSum(long x) {
        long s = 0;
        while (x > 0) {
            s += x % 10;
            x /= 10;
        }
        return s;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        long a = Long.parseLong(br.readLine().trim());
        long b = Long.parseLong(br.readLine().trim());
        while (a != b) {
            if (a < b) a += digitSum(a);
            else b += digitSum(b);
        }
        System.out.println(a);
    }
}

File: src/contests/cg/puzzle/swift/the-river-i-.swift
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

func digitSum(_ x0: Int) -> Int {
    var x = x0
    var s = 0
    while x > 0 {
        s += x % 10
        x /= 10
    }
    return s
}

var a = Int(readLine()!)!
var b = Int(readLine()!)!
while a != b {
    if a < b {
        a += digitSum(a)
    } else {
        b += digitSum(b)
    }
}
print(a)
